File: C/Sully.c
#include <stdio.h>
#include <stdlib.h>
#include <fcntl.h>
#include <unistd.h>

int main(void)
{
	int i = 5;
	int fd;
	char f1[20];
	char f2[20];
	char command[64];
	FILE *fp;
	if (i <= 0)
		return 0;
	sprintf(f1, "Sully_%d.c", i);
	if ((fd = open(f1, O_RDONLY)) > -1)
	{
		i--;
		close(fd);
	}
	sprintf(f2, "Sully_%d.c", i);
	if (!(fp = fopen(f2, "w")))
		return 0;
	char *s = "#include <stdio.h>%1$c#include <stdlib.h>%1$c#include <fcntl.h>%1$c#include <unistd.h>%1$c%1$cint main(void)%1$c{%1$c%2$cint i = %4$d;%1$c%2$cint fd;%1$c%2$cchar f1[20];%1$c%2$cchar f2[20];%1$c%2$cchar command[64];%1$c%2$cFILE *fp;%1$c%2$cif (i <= 0)%1$c%2$c%2$creturn 0;%1$c%2$csprintf(f1, %3$cSully_%%d.c%3$c, i);%1$c%2$cif ((fd = open(f1, O_RDONLY)) > -1)%1$c%2$c{%1$c%2$c%2$ci--;%1$c%2$c%2$cclose(fd);%1$c%2$c}%1$c%2$csprintf(f2, %3$cSully_%%d.c%3$c, i);%1$c%2$cif (!(fp = fopen(f2, %3$cw%3$c)))%1$c%2$c%2$creturn 0;%1$c%2$cchar *s = %3$c%5$s%3$c;%1$c%2$cfprintf(fp, s, 10, 9, 34, i, s);%1$c%2$cfclose(fp);%1$c%2$csprintf(command, %3$cclang -Wall -Wextra -Werror Sully_%%1$d.c -o Sully_%%1$d ; ./Sully_%%1$d%3$c, i);%1$c%2$csystem(command);%1$c}%1$c";
	fprintf(fp, s, 10, 9, 34, i, s);
	fclose(fp);
	sprintf(command, "clang -Wall -Wextra -Werror Sully_%1$d.c -o Sully_%1$d ; ./Sully_%1$d", i);
	system(command);
}

File: PHP/Sully.php
#!/usr/bin/php
<?php

$x = 5;
$str = '#!/usr/bin/php%1$s<?php%1$s%1$s$x = %5$d;%1$s$str = %3$c%4$s%3$c;%1$sif ($x > 0)%1$s{%1$s%2$c$file = @fopen(%3$cSully_%3$c.($x).%3$c.php%3$c, %3$cr%3$c);%1$s%2$cif ($file)%1$s%2$c{%1$s%2$c%2$c$x--;%1$s%2$c%2$cfclose($file);%1$s%2$c}%1$s%2$c$file = fopen(%3$cSully_%3$c.($x).%3$c.php%3$c, %3$cw%3$c);%1$s%2$cif ($file)%1$s%2$c{%1$s%2$c%2$cfprintf($file, $str, PHP_EOL, 9, 39, $str, $x);%1$s%2$c%2$cfclose($file);%1$s%2$c%2$c$fakefile = fopen(%3$cFakeSully%3$c.($x).%3$c.php%3$c, %3$cw%3$c);%1$s%2$c%2$cif ($fakefile)%1$s%2$c%2$c{%1$s%2$c%2$c%2$cfprintf($fakefile, $str, PHP_EOL, 9, 39, $str, $x);%1$s%2$c%2$c%2$cfclose($fakefile);%1$s%2$c%2$c}%1$s%2$c%2$csystem(%3$cphp Sully_%3$c.($x).%3$c.php%3$c);%1$s%2$c}%1$s}%1$s?>%1$s';
if ($x > 0)
{
	$file = @fopen('Sully_'.($x).'.php', 'r');
	if ($file)
	{
		$x--;
		fclose($file);
	}
	$file = fopen('Sully_'.($x).'.php', 'w');
	if ($file)
	{
		fprintf($file, $str, PHP_EOL, 9, 39, $str, $x);
		fclose($file);
		$fakefile = fopen('FakeSully'.($x).'.php', 'w');
		if ($fakefile)
		{
			fprintf($fakefile, $str, PHP_EOL, 9, 39, $str, $x);
			fclose($fakefile);
		}
		system('php Sully_'.($x).'.php');
	}
}
?>
